### ISSUE TITLE
Error preventing new_image_month_stats.txt from being created by smart_labeling.py

### ISSUE THREAD

**Issue by starSweeper** [starSweeper/visibilityMachine#8]

**Comment by starSweeper** [comment]
Could not recreate this issue with small newData1 (testing) folder, will test again with full newData folder once integration to the Java program has been fully tested.

**Comment by starSweeper** [comment]
**Update:** Tomorrow morning, I will run smart_labeling.py on newData (the legit folder, not the testing one), and see if I can recreate the error. It will probably take most of the work day to process the entire folder.

**Comment by starSweeper** [comment]
**Update:** Started running smart_labeling.py with the full folder.... lets see how this goes!
![image](https://user-images.githubusercontent.com/26500355/36491205-6e91c4ee-16e7-11e8-8305-d19a62dee567.png)

**Comment by starSweeper** [comment]
**Update:** Time to go to class... hopefully all continues to run well while I am away.... (note, 2 hours have passed)
![image](https://user-images.githubusercontent.com/26500355/36496182-bdfaebc0-16f4-11e8-815e-8279ba9ffc09.png)

**Comment by starSweeper** [comment]
**Update:** All seems to be going well
![image](https://user-images.githubusercontent.com/26500355/36501028-38227b80-1703-11e8-9050-d71a58a65bac.png)

**Comment by starSweeper** [comment]
**Update:** So it is finished, and as usual the error came at the end of the list when attempting to create the new_image_month_stats.txt file.

![image](https://user-images.githubusercontent.com/26500355/36503849-c712e7b4-170b-11e8-9005-e80f79d831e1.png)

**Comment by starSweeper** [comment]
**Update:** What I don't understand is, all the images are read, processed, and their data is stored during the 3-4 hour plus long "Images Processed" countdown. LONG before any mention of new_image_month_stats.txt. Yet there seems to be some issue reading JPEG files while writing attempting write to that txt file! Before that, as the problem seems to appear when we try to simply form the string that will be written to that txt file! And for some reason it also only happens with the long new images folder....

**Comment by starSweeper** [comment]
**Update:** Running this again, but with some added print statements:
```
 for obj in i_objects:
        print(obj)
        print(len(i_objects))
        month_group[obj.month - 1].append(obj)  # This is were I get a list index out of range error
```

**Comment by starSweeper** [comment]
**Update:** This issue helps illustrate a big problem with this entire project in general -- it takes WAAAAY too long to process images!! It takes 3-4 hours to process about 5,000 images, but it only takes a few seconds (if that) to create the 90,000+ line JSON file that records the processed information! This is not a problem specific to smart_labeling.py, it happens anytime we need to process an image using Python. After this smart labeling mini-project is over, speeding up processing time needs to be a high priority item. It's a real time waster. If it's this slow running on a Windows 10 machine, imagine how much slower it will be on a Rapsberry Pi???

**Comment by starSweeper** [comment]
**Update:** Putting this on hold again. I can't keep testing this program and the Java program at the same time. Not ever since the Java program saves your progress.

**Comment by starSweeper** [comment]
**Update:** I have a suspicion that this is somehow connected to the computer going to sleep while processing images. I will take my work lap top home with me tonight and see if not allowing the computer to go to sleep makes any difference.

**Comment by starSweeper** [comment]
**Update:** This issue is holding the whole project up, so I added the SUPERHIGHPRIORITY tag. It's kind of weird having that tag, and the on hold tag, but the on hold tag is used whenever I cannot currently work on an issue, and it is pretty often that I cannot work on this particular issue because of how long it takes.

**Comment by starSweeper** [comment]
**Update:** Testing without interruptions now.

**Comment by starSweeper** [comment]
**Update:** It is very close to being done processing images, but I accidentally let the laptop fall asleep for a few seconds. After this test I will run another just to be safe.

**Comment by starSweeper** [comment]
**Update:** Yup. Got the error again. I'm running again to be sure (this time I took steps to guarantee no interruptions), but I don't think that's the issue.

**Comment by starSweeper** [comment]
**Update:** Great news! The issue was with one of the images. There was a typo in its name, so that there was one extra character in the date. This added an extra character to the month and made it a number above 12, which gave us an index error when we tried assigning the image to the proper month. The premature end of JPEG file errors are still there, but those can have their own separate issue.

I won't close this issue until I have added a check to make sure all the images have properly formatted dates. One that happens BEFORE the images take several hours to process...
